sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -2524,12 +2656,15 @@ def run(self):
 
         if self.sfp_error_event.is_set():
             sys.exit(SFP_SYSTEM_ERROR)
+        elif generate_sigabrt is True:
+            self.log_notice("Sending SIGABRT to xcvrd!")
```

Comment:
Why do we send a SIGABRT here?

Reply:
This was done for adhering to the below comment
https://github.com/sonic-net/sonic-platform-daemons/pull/377#issuecomment-1593737539

Reply:
In a normal reboot/config reload flow, is it possible to run into this case:
1. xcvrd has started and called self.init (xcvrd has initialized PROC_INFO table)
2. swss starts after and clears PROC_INFO table
3. xcvrd runs into while loop to select PROC_INFO table later, but it does not get the delete event

In such case, PROC_INFO table is empty, but xcvrd has already called notify_media_settings. The state is incorrect.

Reply:
In this case, starting of swss will not clear PROC_INFO table. Only if swss crashes, PROC_INFO table will be cleared.

Also, xcvrd will initialize PROC_INFO table after port config is done. The CmisManagerTask and SfpStateUpdateTask threads also check for port config done before performing their respective initialization.
Following is the function call from xcvrd.init() to ensure port config done is set prior to performing further actions.
https://github.com/sonic-net/sonic-platform-daemons/blob/81048cd1fb63da9fdca567aa863c7d1d43156ed3/sonic-xcvrd/xcvrd/xcvrd.py#L2429